clc/modules/object-storage/src/main/java/com/eucalyptus/objectstorage/ObjectStorageGateway.java

```diff
@@ -630,6 +630,12 @@ public CreateBucketResponseType createBucket(final CreateBucketType request) thr
         throw new InvalidBucketNameException(request.getBucket());
       }
 
+      if (Principals.isFakeIdentify(canonicalId)) {
```

Comment:
Because if we don't check it here, the operationAllowed() will return true, which we don't want in this specific case. In all other cases, we don't check for fake identity because we want to treat the anonymous (nobody) account/user as a first-class Principal like any other.